devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -0,0 +1,711 @@
+          static_cast<SensorPtr<sensor::ISensor>>(sensegloveJointSensor));
+    break;
+  }
+  default: {
+    return {};
+  }
+  }
+
+  return outVec;
+}
+
+wearable::ElementPtr<const wearable::actuator::IActuator>
+HapticGlove::getActuator(const wearable::actuator::ActuatorName name) const {
+  wearable::VectorOfElementPtr<const wearable::actuator::IActuator> actuators =
+      getAllActuators();
+
+  for (const auto &a : actuators) {
+    if (a->getActuatorName() == name) {
+      return a;
+    }
+  }
+  yWarning() << LogPrefix << "User specified actuator name <" << name
+             << "> not found";
+  return nullptr;
+}
+
+wearable::VectorOfElementPtr<const wearable::actuator::IActuator>
+HapticGlove::getActuators(const wearable::actuator::ActuatorType aType) const {
+  wearable::VectorOfElementPtr<const wearable::actuator::IActuator> outVec;
+  switch (aType) {
+  case wearable::actuator::ActuatorType::Haptic: {
+    outVec.reserve(m_pImpl->sensegloveHapticActuatorVector.size());
+    for (const auto &hapticActuator : m_pImpl->sensegloveHapticActuatorVector) {
+      outVec.push_back(
+          static_cast<ElementPtr<actuator::IActuator>>(hapticActuator));
+    }
+    break;
+  }
+  default: {
+    return {};
+  }
+  }
+  return outVec;
+}
+
+wearable::SensorPtr<const wearable::sensor::IVirtualLinkKinSensor>
+HapticGlove::getVirtualLinkKinSensor(
+    const wearable::sensor::SensorName name) const {
+
+  if (!m_pImpl->isAvailable(name,
+                            m_pImpl->gloveData.humanLinkSensorNameIdMap)) {
+    yError() << LogPrefix << "Invalid sensor name " << name;
+    return nullptr;
+  }
+
+  // Return a shared point to the required sensor
+  return dynamic_cast<
+      wearable::SensorPtr<const wearable::sensor::IVirtualLinkKinSensor> &>(
+      *m_pImpl->sensegloveLinkSensorVector
+           [m_pImpl->gloveData.humanLinkSensorNameIdMap[name]]);
+}
+
+wearable::SensorPtr<const wearable::sensor::IVirtualJointKinSensor>
+HapticGlove::getVirtualJointKinSensor(
+    const wearable::sensor::SensorName name) const {
+  if (!m_pImpl->isAvailable(name,
+                            m_pImpl->gloveData.humanJointSensorNameIdMap)) {
+    yError() << LogPrefix << "Invalid sensor name " << name;
+    return nullptr;
+  }
+
+  // Return a shared point to the required sensor
+  return dynamic_cast<
+      wearable::SensorPtr<const wearable::sensor::IVirtualJointKinSensor> &>(
+      *m_pImpl->sensegloveJointSensorVector
+           [m_pImpl->gloveData.humanJointSensorNameIdMap[name]]);
+}
+
+wearable::ElementPtr<const wearable::actuator::IHaptic>
+HapticGlove::getHapticActuator(const actuator::ActuatorName name) const {
+  // Check if user-provided name corresponds to an available actuator
+  if (!m_pImpl->isAvailable(name,
+                            m_pImpl->gloveData.humanHapticActuatorNameIdMap)) {
+    yError() << LogPrefix << "Invalid actuator name " << name;
+    return nullptr;
+  }
+
+  return dynamic_cast<
```

Comment:
Just checking, is the `dynamic_cast` necessary? Downcast should be done automatically. If it is necessary, and your sure that the original data can be successfully cast, I'd recommend using `static_cast` that has less runtime cost.

Reply:
changed them to static cast.